### ISSUE TITLE
RFC: Branching strategy update

### ISSUE THREAD

**Issue by axelson** [elixir-lsp/elixir-ls#623]
Since we're likely to get a website soon (see #619 and the example site http://danpetrov.xyz/elixir-ls.github.io/) I've been thinking about lightweight ways to keep the website in sync with the released version of the extension. So I'd like to propose that we use a `dev` branch as the default branch on GitHub. That means that all branches by default will use `dev` as the base branch.

Additionally I'd like to use `main` as the mainline branch since that's the new default.

I'd like to know if anyone has feedback or thoughts about these changes.

cc: @lukaszsamson @msaraiva

**Comment by lukaszsamson** [comment]
Couldn't the website be built from tagged commits?

**Comment by DaniruKun** [comment]
I have to agree with @lukaszsamson - a trunk-based approach in this case is much simpler and avoids the complexity of having to now deal with an additional branch. If the concern is about the website versioning, there is https://squidfunk.github.io/mkdocs-material/setup/setting-up-versioning/ . For example, for each new release tag, a new, `latest` version is built, while also allowing users to see docs for older versions.

However imho for a mid-sized project like this which is still not complete, I would instead encourage users to upgrade to the latest version.

**Comment by axelson** [comment]
Ah, that is a good point. Yeah let's build the website from the tagged commits :+1:
